tools/test_tpm_ek_chain.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+WORKDIR="$(mktemp -d)"
+declare -i MAX_CERT_SIZE=0
+declare -i CHAIN_LEN=0
+declare -i CHAIN_SIZE=0
+
+function cleanup {
+	rm -rf $WORKDIR
+	exit
+}
+
+# tpm2-tools changed command and option names. This script was tested with
+# versions 5.2 and 3.0.2, and may not work on different versions. Contibutions
+# welcome.
+which tpm2_pcrread > /dev/null
+if [ $? -eq 0 ]; then
+	# 5.2
+	pcrread="tpm2_pcrread"
+	nvread="tpm2_nvread -C o"
+else
+	# 3.0.2
+	pcrread="tpm2_pcrlist"
+	nvread="tpm2_nvread -x"
+fi
+
+# Do we have working TPM2.0?
+$pcrread >/dev/null 2>&1
+if [ $? -ne 0 ]; then
+	echo "Command execution failed, is TPM2.0 available?"
+	cleanup
+fi
+
+$nvread 0x01C00002 > $WORKDIR/0.der 2>/dev/null
+if [ $? -ne 0 ]; then
+	echo "TPM doesn't have EK certificate at index 0x01C00002"
+	cleanup
+fi
+
+while true; do
+	cert=$WORKDIR/$CHAIN_LEN
+	ca_url=`openssl x509 -inform der -in $cert.der -text -noout | grep "CA Issuers" | cut -d : -f 2-`
+	size=`stat -c %s $cert.der`
+
+	echo "Certificate $CHAIN_LEN size: $size bytes"
+	((CHAIN_LEN++))
+	CHAIN_SIZE+=$size
+	if [ $size -gt $MAX_CERT_SIZE ]; then
+		MAX_CERT_SIZE=$size
+	fi
+
+	# 'openssl verify' can't use DER, must convert to PEM
+	openssl x509 -inform der -in $cert.der -outform pem -out $cert.pem
+	openssl verify -CAfile $cert.pem $cert.pem >/dev/null 2>&1
+	if [ $? -eq 0 ]; then
+		echo "Certificate is self-signed, assuming it is root"
+		break
+	fi
+
+	wget -q -O $WORKDIR/$CHAIN_LEN.der $ca_url
+	if [ $? -ne 0 ]; then
+		echo "Can't download CA certificate from '$ca_url'"
+		cleanup
+	fi
+done
+
+echo
+echo "Chain length: $CHAIN_LEN"
+echo "Total chain size: $CHAIN_SIZE bytes"
+echo "Size of the biggest certificate: $MAX_CERT_SIZE bytes"
+
+echo
+echo "Verifying whole chain:"
+
+ca_files=$(find $WORKDIR -name '*.pem' ! -name 0.pem)
+
+# Pipe EK cert to hide name given by mktemp and let openssl speak for itself
+cat $WORKDIR/0.pem | openssl verify -CAfile <(cat $ca_files)
+
+cleanup
```

Comment:
@krystian-hebel I know you do not like it, but in the issue you could also provide one-liner to make it as simple as possible to download and execute the script. Or at least a set of command to download and execute. Right now the link is hidden in the text and assumes I know how to download and execute script, and have enough determination to do it by myself. Copying commands is alwasy a bit faster.

Reply:
Agree, but let's merge this first so I can use the final URL. I'll address rest of the comments first thing tomorrow.